### PR TITLE
fix: don't terminate when got normal error

### PR DESCRIPTION
**Pull request by TheWaWaR**

### CODE REVIEW

File: c/polyjuice.h [hunk 1]
```diff
@@ -567,20 +577,24 @@ struct evmc_result call(struct evmc_host_context* context,
...
   } else {
     ret = handle_message(gw_ctx, context->from_id, context->to_id, &context->destination, msg, &res);
+    if (is_fatal_error(ret)) {
+      context->error_code = ret;
+    }
     if (ret != 0) {
       ckb_debug("inner call failed (transfer/contract call contract)");
-      context->error_code = ret;
-      res.status_code = EVMC_REVERT;
+      res.status_code = EVMC_INTERNAL_ERROR;
```
Comment (jjyr) [inline c/polyjuice.h:597]: What happend if we failed to call another contract? shouldn't it be a EVMC_REVERT status?

File: c/sudt_contracts.h [hunk 2]
```diff
@@ -49,20 +46,25 @@ int balance_of_any_sudt(gw_context_t* ctx,
...
   }
   *output_size = 32;
   memset(*output, 0, 32);
+
+  evmc_address address = *((evmc_address *)(input_src + 32 + 12));
+  uint128_t balance;
+  ret = sudt_get_balance(ctx, sudt_id, POLYJUICE_SHORT_ADDR_LEN, address.bytes, &balance);
+  if (ret == GW_ERROR_NOT_FOUND) {
+    debug_print_int("sudt account not found", sudt_id);
+    return 0;
```
Comment (jjyr) [inline c/sudt_contracts.h:63]: output is uninitialized in this branch

**Review by jjyr — CHANGES_REQUESTED** [review]

File: c/sudt_contracts.h (hunk 2 above)
Comment (TheWaWaR) [reply c/sudt_contracts.h:63]: Line:59 => `memset(*output, 0, 32);`

**Review by jjyr — CHANGES_REQUESTED** [review]
I suggest to remove smt.h, using `ctx->_internal_store_raw(k, v)` to instead `smt_state_insert`, using `ctx->_internal_load_raw(k,v)` to instead `smt_state_fetch`

File: c/polyjuice.h [hunk 3]
```diff
@@ -491,12 +517,13 @@ void selfdestruct(struct evmc_host_context* context,
   int ret;
   if (memcmp(beneficiary->bytes, context->destination.bytes, 20) == 20) {
```
Comment (jjyr) [inline c/polyjuice.h:518]: Is the condition right?? memcmp(...) == **20**?

**Review by jjyr — CHANGES_REQUESTED** [review]

File: c/other_contracts.h [hunk 4]
```diff
@@ -41,7 +37,8 @@ int recover_account(gw_context_t* ctx,
                     const size_t input_size,
                     uint8_t** output, size_t* output_size) {
   if (input_size < 128) {
-    return ERROR_RECOVER_ACCOUNT;
+    debug_print_int("input size too small", input_size);
+    return 0;
```
[the remark this answers is not in the document]
Comment (TheWaWaR) [reply c/other_contracts.h:41]: The `output` default is `NULL` and the `output_size` default is 0.

File: c/polyjuice.h [hunk 5]
```diff
@@ -73,7 +84,7 @@ int gw_increase_nonce(gw_context_t *ctx, uint32_t account_id, uint32_t *new_nonc
 #ifdef GW_GENERATOR
   ret = syscall(GW_SYS_STORE, nonce_key, nonce_value, 0, 0, 0, 0);
 #else
-  ret = gw_state_insert(&ctx->kv_state, nonce_key, nonce_value);
+  ret = smt_state_insert(&ctx->kv_state, nonce_key, nonce_value);
 #endif
   if (ret != 0) {
     return ret;
```
[the remark this answers is not in the document]
Comment (TheWaWaR) [reply c/polyjuice.h:90]: ok

File: c/polyjuice.h (hunk 3 above)
Comment (TheWaWaR) [reply c/polyjuice.h:518]: Oops.

Comment (TheWaWaR) [reply c/polyjuice.h]: This check is not necessary, remove it.

**Review by Flouse — APPROVED** [review]

**Review by jjyr — APPROVED** [review]

**Pull request merged**